satpy/composites/__init__.py

```diff
@@ -549,71 +549,104 @@ def _insert_palette_colors(channels, palette):
 
 
 class DayNightCompositor(GenericCompositor):
-    """A compositor that blends a day data with night data."""
+    """A compositor that blends day data with night data.
+    Using the `day_night` flag it is also possible to provide only a day product
+    or only a night product and mask out (make transparent) the opposite portion
+    of the image (night or day). See the documentation below for more details.
+    """
 
-    def __init__(self, name, lim_low=85., lim_high=88., **kwargs):
+    def __init__(self, name, lim_low=85., lim_high=88., day_night="day_night", **kwargs):
         """Collect custom configuration values.
 
         Args:
             lim_low (float): lower limit of Sun zenith angle for the
                              blending of the given channels
             lim_high (float): upper limit of Sun zenith angle for the
                              blending of the given channels
+            day_night (string): "day_night" means both day and night portions will be kept
+                                "day_only" means only day portion will be kept
+                                "night_only" means only night portion will be kept
 
         """
         self.lim_low = lim_low
         self.lim_high = lim_high
+        self.day_night = day_night
         super(DayNightCompositor, self).__init__(name, **kwargs)
 
     def __call__(self, projectables, **kwargs):
         """Generate the composite."""
-        projectables = self.match_data_arrays(projectables)
-
-        day_data = projectables[0]
-        night_data = projectables[1]
+        # At least one composite is requested.
+        foreground_data = projectables[0]
```

Comment:
`match_data_arrays` should *always* be called. You can do `projectables = self.match_data_arrays(projectables)` just above this line. `match_data_arrays` checks a lot of various things and modifies the provided DataArrays as needed to make them compatible with the compositor or with the rest of Satpy.

Reply:
Well when I tested `true_color_with_night_ir_alpha` I got errors saying `Dimension 'bands' size mismatch: 3 != 4` with `match_data_arrays`. This even occurred under original version. I didn't know what happened...

Reply:
Is this something related with #1821 and #1813 ?

Reply:
Yes, this is exactly #1821. I'm trying to finish #1815 and then fix #1821. Everything is just taking longer than I expected. One work around is to switch to an older version of xarray (`<0.19.0` if I remember correctly).

Reply:
Ok so I'll just put `match_data_arrays` back here.